src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -680,7 +680,7 @@ protected virtual ExpressionToken NextTokenImplementation(out Exception error)
                         }
 
                         this.ParseIdentifier();
-                        t = ExpressionTokenKind.ParameterAlias;
+                        t = this.parsingFunctionParameters && !(this.ExpressionText[0] == '@' && this.ExpressionText.Contains(".")) ? ExpressionTokenKind.ParameterAlias : ExpressionTokenKind.Identifier;

```

Comment:
nit: multiline would be easier to read.

Reply:
indeed.  fixed.

---
In reply to: [167023378](https://github.com/OData/odata.net/pull/1063#discussion_r167023378) [](ancestors = 167023378)